src/reducer.js

```diff
@@ -639,6 +684,89 @@ export const onWalletBestBlockUpdate = (state, action) => {
   };
 };
 
+// Push notification
+
+/**
+ * @param {{ deviceId: string, settings: { enabled, showAmountEnabled }, hasBeenEnabled: boolean }} action
+ */
+export const onPushInit = (state, action) => {
+  const { deviceId, settings, hasBeenEnabled } = action.payload;
+  return ({
+    ...state,
+    pushNotification: {
+      ...state.pushNotification,
+      ...settings,
+      deviceId,
+      hasBeenEnabled,
+    },
+  });
+};
+
+export const onPushUpdateDeviceId = (state, action) => {
+  const { deviceId } = action.payload;
+  return ({
+    ...state,
+    pushNotification: {
+      ...state.pushNotification,
+      deviceId,
+    },
+  });
+};
+
+export const onPushApiReady = (state) => ({
+  ...state,
+  pushNotification: {
+    ...state.pushNotification,
+    apiStatus: PUSH_API_STATUS.READY,
+  },
+});
+
+/**
+ * @param {{enabled: boolean, hasBeenEnabled: boolean, enabledAt: number }} action
+ */
+export const onPushRegisterSuccess = (state, action) => {
+  const { enabled, hasBeenEnabled, enabledAt } = action.data;
+  return ({
+    ...state,
+    pushNotification: {
+      ...state.pushNotification,
+      apiStatus: PUSH_API_STATUS.READY,
+      enabled,
+      hasBeenEnabled,
+      enabledAt,
+    },
+  });
+};
+
+export const onPushRegisterFailed = (state) => ({
+  ...state,
+  pushNotification: {
+    ...state.pushNotification,
+    apiStatus: PUSH_API_STATUS.FAILED,
+  },
+});
+
+/**
+ * @param {{payload: {enabled, showAmountEnabled}}} action
+ */
+export const onPushUpdateSuccess = (state, { payload: { enabled, showAmountEnabled } }) => ({
+  ...state,
+  pushNotification: {
+    ...state.pushNotification,
+    apiStatus: PUSH_API_STATUS.READY,
+    enabled,
+    showAmountEnabled,
+  },
+});
+
+export const onPushUpdateFailed = (state) => ({
```

Comment:
Why do you need `onPushUpdateFailed` and `onPushRegisterFailed` with the same code? Can't we have a `onPushAPIFailed`?

Reply:
Good point. I kept them separated hopping they could evolve differently, but this is not the case indeed. I will join them.